src/google/protobuf/compiler/cpp/cpp_helpers.cc

```diff
@@ -70,7 +70,11 @@ const char* const kKeywordList[] = {
   "return", "short", "signed", "sizeof", "static", "static_assert",
   "static_cast", "struct", "switch", "template", "this", "thread_local",
   "throw", "true", "try", "typedef", "typeid", "typename", "union", "unsigned",
-  "using", "virtual", "void", "volatile", "wchar_t", "while", "xor", "xor_eq"
+  "using", "virtual", "void", "volatile", "wchar_t", "while", "xor", "xor_eq",
+  // OBJECTIVE C KEYWORDS AND MACROS
+  "__autoreleasing", "__strong", "__unsafe_unretained", "__weak", "id", "Class",
+  "_cmd", "BOOL", "BSD", "bycopy", "byref", "IMP", "In", "inout", "Nil", "nil",
+  "NO", "oneway", "out", "SEL", "self", "super", "YES"
```

Comment:
This will likely to break existing clients, especially those all caps ones. I did a quick code search internally all the caps one are already being used as enum values: BOOL, BSD, YES, NO and SEL.